contracts/test/v0.8/dev/DerivedPriceFeed.test.ts

```diff
@@ -0,0 +1,141 @@
+import { ethers } from 'hardhat'
+import { BigNumber, ContractFactory } from 'ethers'
+import { expect } from 'chai'
+import { describe } from 'mocha'
+
+describe('DerivedPriceFeed', () => {
+  let mockAggFactory: ContractFactory
+  let derivedFeedFactory: ContractFactory
+  before(async () => {
+    const accounts = await ethers.getSigners()
+    mockAggFactory = await ethers.getContractFactory(
+      'src/v0.7/tests/MockV3Aggregator.sol:MockV3Aggregator',
+      accounts[0],
+    )
+    derivedFeedFactory = await ethers.getContractFactory(
+      'src/v0.8/dev/DerivedPriceFeed.sol:DerivedPriceFeed',
+      accounts[0],
+    )
+  })
+
+  it('reverts on getRoundData', async () => {
+    let base = await mockAggFactory.deploy(8, 10e8) // Price = 10
+    let quote = await mockAggFactory.deploy(8, 5e8) // Price = 5
+
+    let derived = await derivedFeedFactory.deploy(
+      base.address,
+      quote.address,
+      8,
+    )
+
+    await expect(derived.getRoundData(1)).to.be.reverted
+  })
+
+  it('returns decimals', async () => {
+    let base = await mockAggFactory.deploy(8, 10e8) // Price = 10
+    let quote = await mockAggFactory.deploy(8, 5e8) // Price = 5
+
+    let derived = await derivedFeedFactory.deploy(
+      base.address,
+      quote.address,
+      9,
+    )
+
+    await expect(await derived.decimals()).to.equal(9)
+  })
+
+  describe('calculates price', async () => {
+    it('when all decimals are the same', async () => {
+      let base = await mockAggFactory.deploy(8, 10e8) // 10
+      let quote = await mockAggFactory.deploy(8, 5e8) // 5
+
+      let derived = await derivedFeedFactory.deploy(
+        base.address,
+        quote.address,
+        8,
+      )
+
+      await expect((await derived.latestRoundData()).answer).to.equal(
+        BigNumber.from('200000000' /* 2 */),
```

Comment:
Super small nit: can we use scientific notation in the string instead? `2e8`. Easier to read IMO. Similar comments for the rest of the tests.

Reply:
BigNumber.from doesn't support that, but I got rid of it where I could and replaced with scientific notation